### DISCUSSION TITLE
Error en la pokedex

### DISCUSSION THREAD

**Discussion by AndresAVargas** [LaunchX-InnovaccionVirtual/FrontEnd-Mision#525] [Q&A]
Hola compañeros, no se si puedan ayudarme, estoy haciendo la pokedex, cuando busco 1 pokemon que tiene 2 tipos me salen los resultados pero al buscar 1 tipo me aparece esto
![image](https://user-images.githubusercontent.com/88987587/160185444-09480c8e-e1bb-4130-9e12-0e848dfa1dfc.png)
Ya lo intente solucionar pero creo que lo estoy haciendo mal, no se si alguien me pueda ayudar, mi codigo es este
![image](https://user-images.githubusercontent.com/88987587/160186079-c9f71dd5-eab5-4161-a756-338ef0dbfb1a.png)
![image](https://user-images.githubusercontent.com/88987587/160186100-b60cd7a3-731c-4c6d-8dfc-a2a2aa752564.png)
![image](https://user-images.githubusercontent.com/88987587/160186113-323a47a7-4c2e-4c61-94cc-660f650a133d.png)
![image](https://user-images.githubusercontent.com/88987587/160186126-773a9d68-1ef1-40ab-b088-64d205bc5ae4.png)

**Comment by khisus19** [comment]
Amigo yo también tuve exactamente el mismo problema y lo solucioné así

```
if (data.types.length == 2) {
            let pokeTipo2 = data.types[1].type.name;
            let tipo2 = capitalizarPrimeraLetra(pokeTipo2);
            
            document.getElementById("pokeTipo2").innerHTML = `${tipo2}`;
}
else document.getElementById("pokeTipo2").innerHTML = `-`;
```
El condicional dice que si el arreglo data.types tiene dos elementos entonces muestre el tipo2 . Si no tiene dos elementos muestre un "-"

Puedes ver mi pokedex aqui: [Pokedex](https://khisus19.github.io/pokedex/)

**Reply by AndresAVargas** [reply]
El condicional lo metiste en el fetch o definiste otra funcion?

**Reply by khisus19** [reply]
Si esta dentro del fetch

Aquí esta mi repo https://github.com/khisus19/pokedex

**Reply by AndresAVargas** [reply]
Ya lo arregle compañero muchas gracias, me sirvio tu ayuda

**Reply by khisus19** [reply]
No hay de que. Estamos para ayudarnos

**Comment by doguedogue** [comment]
y si tiene tres tipos, o cuatro?
lo puedes hacer dinámico para que lea cuantos traiga (o si no trae)

`let tipos = data.types.map(typ => typ.type.name);`

**Reply by AndresAVargas** [reply]
no entendi compañero, me podrias explicar mas?

**Reply by khisus19** [reply]
Según tengo entendido no existen pokemon con más de dos tipos. Y por eso se me ocurrió hacerlo así. Pero muchas gracias por compartir esa forma. Soy muy novato y los recorridos de arrays no los domino aún
